dialdbg/src/main.rs

```diff
@@ -0,0 +1,239 @@
+        secret.to_string(),
+    );
+
+    // `connect` may propagate an error here; log the error with a prefix so we can still
+    // process logs and not immediately return from the main function.
+    if let Err(e) = dial::DialOptions::builder()
+        .uri(uri)
+        .with_credentials(creds)
+        .disable_webrtc()
+        .allow_downgrade()
+        .connect()
+        .await
+    {
+        log::error!("{}: {e}", strings::DIAL_ERROR_PREFIX);
+    }
+}
+
+async fn dial_webrtc(uri: &str, secret: &str) -> Option<stats::StatsReport> {
+    let creds = dial::RPCCredentials::new(
+        None,
+        "robot-location-secret".to_string(),
+        secret.to_string(),
+    );
+
+    // `connect` may propagate an error here; log the error with a prefix so we can still
+    // process logs and not immediately return from the main function. Assuming there was
+    // no error, return the stats report of the underlying RTCPeerConnection.
+    match dial::DialOptions::builder()
+        .uri(uri)
+        .with_credentials(creds)
+        .allow_downgrade()
+        .connect()
+        .await
+    {
+        Ok(c) => match c {
+            ViamChannel::WebRTC(c) => Some(c.get_stats().await),
+            _ => None,
+        },
+        Err(e) => {
+            log::error!("{}: {e}", strings::DIAL_ERROR_PREFIX);
+            None
+        }
+    }
+}
+
+fn output_connection_stats(stats: stats::StatsReport, out: &mut Box<dyn io::Write>) -> Result<()> {
+    // StatsReport contains 13 types of stat reports; there may be more relevant stats to print
+    // here, but for now I have stuck with only printing the candidates.
+    writeln!(out, "\nnominated ICE candidates:\n")?;
+    let now = Instant::now();
+    for value in stats.reports.into_values() {
+        match value {
+            stats::StatsReportType::LocalCandidate(lcs) => {
+                writeln!(out, "\tlocal ICE candidate:")?;
+                writeln!(out, "\t\tIP address: {}", lcs.ip)?;
+                writeln!(out, "\t\tport: {}", lcs.port)?;
+                writeln!(
+                    out,
+                    "\t\tnominated {:#?} ago",
+                    now.duration_since(lcs.timestamp)
+                )?;
+                writeln!(out, "\t\trelay protocol: {}", lcs.relay_protocol)?;
+                writeln!(out, "\t\tnetwork type: {}", lcs.network_type)?;
+            }
+            stats::StatsReportType::RemoteCandidate(rcs) => {
+                writeln!(out, "\tremote ICE candidate:")?;
+                writeln!(out, "\t\tIP address: {}", rcs.ip)?;
+                writeln!(out, "\t\tport: {}", rcs.port)?;
+                writeln!(
+                    out,
+                    "\t\tnominated {:#?} ago",
+                    now.duration_since(rcs.timestamp)
+                )?;
+                writeln!(out, "\t\trelay protocol: {}", rcs.relay_protocol)?;
+                writeln!(out, "\t\tnetwork type: {}", rcs.network_type)?;
+            }
```

Comment:
nit
```suggestion
                        stats::StatsReportType::LocalCandidate(ref cand)
            | stats::StatsReportType::RemoteCandidate(ref cand) => {
                let remote_or_local = if let stats::StatsReportType::LocalCandidate(_) = value {
                    "remote"
                } else {
                    "local"
                };
                writeln!(out, "\t{} ICE candidate:", remote_or_local)?;
                writeln!(out, "\t\tIP address: {}", cand.ip)?;
                writeln!(out, "\t\tport: {}", cand.port)?;
                writeln!(
                    out,
                    "\t\tnominated {:#?} ago",
                    now.duration_since(cand.timestamp)
                )?;
                writeln!(out, "\t\trelay protocol: {}", cand.relay_protocol)?;
                writeln!(out, "\t\tnetwork type: {}", cand.network_type)?;
            }
```

Reply:
Moved local and remote to one arm as described above. The `matches` macro kind of scared me haha, but if you feel strongly it's a better solution, @stuqdog, I can look more into it.

Reply:
@stuqdog just for my own edification, what would be better about the macro?

Reply:
@benjirewis I don't feel strongly, this looks fine!
@edaniels it's not a big difference, but currently we have the catch-all no-op case at the bottom, which we can avoid with `if matches!(<var>, local | remote) { ... }`. Nothing major, just a bit cleaner imo!

Reply:
Yeah I couldn't find a way to use the `matches` macro _and_ extract the actual candidate from the `LocalCandidate` / `RemoteCandidate` enum. i.e. couldn't do `if matches!(report_value, LocalCandidate(cand) | RemoteCandidate(cand) { ...access cand })`. Maybe I'm missing something, though.